bugalink-backend/users/views.py

```diff
@@ -126,4 +126,90 @@ class UserRatingView(viewsets.GenericViewSet,
     serializer_class = UserRatingSerializer
 
     def get(self, request, *args, **kwargs):
-        return self.retrieve(request, *args, **kwargs)
\ No newline at end of file
+        return self.retrieve(request, *args, **kwargs)
+    
+# GET /users/<user_id>/trip-requests/count?status=pending
+# GET /users/<user_id>/trip-requests/count?status=accepted
+# GET /users/<user_id>/trip-requests/count?status=finished # HISTORIAL
+class UserTripCountView(APIView):
+    def get(self, request, id):
+        # If the user is not the same as the one in the URL, return a 403 status code
+        if request.user.id != int(id):
+            return Response(
+                data={
+                    "error": "No tienes permiso para ver los viajes de otro usuario."
+                },
+                status=status.HTTP_403_FORBIDDEN,
+            )
+
+        # Get the status from the query params
+        # If the status is not in the query params, don't filter by it
+        status_param = request.query_params.get("status")
+        status_list = status_param.split(",") if status_param else []
+
+        user = User.objects.get(id=id)
+        # Get the trips from the user. Those are the trips in which the user is a passenger
+        # A trip has a many-to-many relationship with passengers, and we want to see if the
+        # user's passenger id from request.user.id is in that list of passengers
+        # trip_requests_where_user_is_passenger = TripRequest.objects.filter(
+        #     passenger__user=user
+        # )
+
+        trips_where_user_is_passenger = TripRequest.objects.filter(passenger__user=user)
+
+        trips_where_user_is_driver = TripRequest.objects.filter(
+            trip__driver_routine__driver__user=user
+        )
+
+        trips_by_user = trips_where_user_is_passenger | trips_where_user_is_driver
+
+        # Filter the trips based on the status values
+        trips_matching_status = (
+            trips_by_user.filter(status__in=status_list)
+            if status_list
+            else trips_by_user
+        )
+
+        # Return the number of trips with a 200 status code
+        return Response(
+            data= {
+                "numTrips": len(trips_matching_status),
+            }
+        )
+        
+# GET /users/<user_id>/trips/upcoming
```

Comment:
Upcoming trips is already working and integrated using the `/users/1/trip-requests?status=PENDING,ACCEPTED` endpoint, so `/users/1/trips/upcoming` could be redundant. I see that it also filters for those in less than 2 days in the future, that could be useful, but if we do that we can't find our trips that are accepted, for example, for next week

Since tripRequest statuses have been changed as well as trips, the endpoint will probably need to be changed to consider those trips whose status is ACCEPTED but not FINISHED, since FINISHED will not be a status for trip-requests anymore but for trips.

![image](https://user-images.githubusercontent.com/53608787/230433439-17d1eed8-bc8a-4862-85e4-ee55d4b04318.png)

Reply:
The idea of having the upcoming trips function is to be able to show a short list of the upcoming trips in the front page, it's not meant to substitute the already existing function that obtains all trip-requests of a specific status for a user.

I don't understand the second paragraph, I wouldn't say either of the implemented functions makes use of the trip status. If anything, maybe I should add a filter on the "upcoming trips" method, where it checks if the trip-request has been accepted.